### DISCUSSION TITLE
Run part of extension on "host" and part in dev container

### DISCUSSION THREAD

**Discussion by itavero** [microsoft/vscode-discussions#1137] [Extension Development QnA]
I'm looking for a way to start certain processes on the "host" machine and inform extension code in the dev container about the results.
From what I gathered, an extension can have only one [Extension Host](https://code.visualstudio.com/api/advanced-topics/extension-host), so I think for this use case I would need two extensions (one on `local` and one on `remote`).

**A bit of background to this request:**
I'm working on embedded software, for which I have a debug probe connected to my host machine (e.g. `local`) which is physically connected to the device I'm developing software for.
To simplify configuration management, the compiler toolchain will be put in a Dev Container (e.g. the compiler but also GDB).
Before looking into Dev Containers we used `marus25.cortex-debug` for debugging, but it seems currently the only way to use that with our Docker based Dev Containers (running in WSL2 on Windows), would involve annoying mappings and non-user-friendly tools like `usbipd`.

Ideally, I would want a "plug n play" setup, like we had before. When using a SEGGER J-Link probe, this can be achieved relatively easy by starting a [J-Link Remote Server](https://www.segger.com/products/debug-probes/j-link/tools/j-link-remote-server/).
So, I'd like to automate the process of starting the server and then figuring out which IP address (or tunnel) `marus25.cortex-debug` should use (potentially from an additional extension).

**Comment by tamuratak** [comment]
Although I know nothing about `marus25.cortex-debug`, the following comment would be helpful:

- https://github.com/Marus/cortex-debug/issues/467#issuecomment-1809068641

Since VS Code doesn't directly provide a communication channel between `local` and `remote` extensions, you have to implement it with the [port forwarding feature](https://code.visualstudio.com/docs/devcontainers/containers#_forwarding-or-publishing-a-port). Typically, you start a WebSocket server on a remote machine and connect to it from a local extension through the port forwarding.

**Comment by abrilurena** [comment]
Hi, @itavero! hope you're doing great. Do you still need help with this?

**Comment by abrilurena** [comment]
Hi, @itavero! it's been a while since last time, so i'm cosing this issue for now. Please feel free to re-open if you need more help ✨
